### PR TITLE
[3pt] PR:  Refactor Calibration Database to use .parquet files

### LINKED ISSUE #894: [13pt] Refactor calibration routine

### PR DESCRIPTION
**Pull request by robgpita-noaa**
Refactor the use of how calibration points are stored, and read for processing Synthetic Rating Curve Adjustments.  A PostgreSQL database is currently used to populate and query points located within a HUC8 based on a spatial query of two tables in the database.  A script (`write_parquet_from_calib_pts.py`) is added to write `.parquet` files containing calibration points for each HUC8 based on two `.gpkg` files containing HUC8 polygons, and calibration points.  Modifications were made to read the calibration points from `.parquet` files. 

Removal of the PostgreSQL database leads to faster post processing times (~5 minutes per run), as well as an increase in ease of use for external collaborators (no need to set up PostgreSQL docker container).

Closes #894, #582

### Additions
- `data/`
    - `write_parquet_from_calib_pts.py`: Script to write `.parquet` files (per HUC8) based on calibration points contained in a .gpkg file associated with each HUC8. 

### Changes  
- `src/`
    - `src_adjust_spatial_obs.py`: Refactor to remove PostgreSQL and use `.parquet` files.
    - `src_roughness_optimization.py`: Line up comments and add newline at EOF. 
    - `bash_variables.env`: Update formatting, and add `{}` to inherited `.env` variables for proper variable expansion in Python scripts.  
- `/config`
    - `params_template.env`: Update comment.
- `fim_pre_processing.sh`: In usage statement, remove references to PostGRES calibration tool.
- `fim_post_processing.sh`: Remove connection to and loading of PostgreSQL database. 
- `.gitignore`: Add newline.
- `README.md`: Remove references to PostGRES calibration tool.

### Removals
- `config/` 
    - `calb_db_keys_template.env`: No longer necessary without PostGRES Database.

- `/tools/calibration-db` : Removed directory including files below. 
    - `README.md`
    - `docker-compose.yml`
    - `docker-entrypoint-enitdb.d/init-db.sh`

### Testing
A `fim_pipeline.sh` run of three HUCS ("12040103 04100003 12090301") was submitted, using the `-debug` flag to `src_adjust_spatial_obs.py`. When overlaid/viewed in QGIS, the intermediate output showed no difference in calibration coefficients between `fim_4_3_11_6`. Outputs can be found here: `/dev_fim_share/foss_fim/outputs/test_hucs_calib_NEW_parquet`.

An Alpha Domain and associated Alpha Tests (`synthesize_test_cases.py` & `eval_plots.py`) was run on this feature branch. There were identical results to `fim_4_3_11_6` in the Aggregate CSI metrics gathered from the 5 benchmark datasets.  Outputs can be found here: `/dev_fim_share/foss_fim/outputs/test_parquet_alpha_domain/`

### Issuer Checklist (For developer use)

_You may update this checklist before and/or after creating the PR. If you're unsure about any of them, please ask, we're here to help! These items are what we are going to look for before merging your code._

- [x] Informative and human-readable title, using the format: `[_pt] PR: <description>`
- [x] Links are provided if this PR resolves an issue, or depends on another other PR
- [x] If submitting a PR to the `dev` branch (the default branch), you have a descriptive [Feature Branch](https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow) name using the format: `dev-<description-of-change>` (e.g. `dev-revise-levee-masking`)
- [x] Changes are limited to a single goal (no scope creep)
- [x] The feature branch you're submitting as a PR is up to date (merged) with the latest `dev` branch
- [x] Changes adhere to [PEP-8 Style Guidelines](https://peps.python.org/pep-0008/)
- [x] Any _change_ in functionality is tested
- [x] Passes all unit tests locally (inside interactive Docker container, at `/foss_fim/`, run: `pytest unit_tests/`)
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future todos are captured in comments
- [x] [CHANGELOG](/docs/CHANGELOG.md) updated with template version number, e.g. `4.x.x.x` 
- [x] [Reviewers requested](https://help.github.com/articles/requesting-a-pull-request-review/)
- [x] Add yourself as an [assignee](https://docs.github.com/en/issues/tracking-your-work-with-issues/assigning-issues-and-pull-requests-to-other-github-users) in the PR  as well as the FIM Technical Lead

### Merge Checklist (For Technical Lead use only)

- [ ] Update [CHANGELOG](/docs/CHANGELOG.md) with latest version number
- [ ] Update the [Citation.cff](/CITATION.cff) file to reflect the latest version number in the [CHANGELOG](/docs/CHANGELOG.md)
- [ ] If applicable, update [README](/README.md) with major alterations

### CODE REVIEW

File: config/params_template.env [hunk 1]
```diff
@@ -40,15 +40,13 @@ export src_subdiv_toggle="True" # Toggle to run composite roughness src routine
 export vrough_suffix="" # text to append to output log and hydrotable file names (use for testing/debugging)
 export vmann_input_file="/data/inputs/rating_curve/variable_roughness/mannings_global_06_12.csv" # input file location with nwm feature_id and channel roughness and overbank roughness attributes
 
-#### apply SRC adjustments using USGS rating curve database (calibration db) ####
+#### apply SRC adjustments using USGS rating curve database (calibration points db) ####
```
Comment (RyanSpies-NOAA) [inline config/params_template.env:43]: This is block is for the USGS rating curve calibration (not the point data base)

File: src/src_adjust_spatial_obs.py [hunk 2]
```diff
@@ -74,7 +66,7 @@ def process_points(args):
     ## Define coords variable to be used in point raster value attribution.
     coords = [(x,y) for x, y in zip(water_edge_df.X, water_edge_df.Y)]
 
-    water_edge_df.to_crs(DEFAULT_FIM_PROJECTION_CRS)
+    water_edge_df = water_edge_df.set_crs(DEFAULT_FIM_PROJECTION_CRS, allow_override=True)
```
Comment (RyanSpies-NOAA) [inline src/src_adjust_spatial_obs.py:69]: `set_crs` assigns the crs without performing a transformation. I think we want to use `to_crs` to insure the input data gets converted to the `DEFAULT_FIM_PROJECTION` if needed. https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoDataFrame.set_crs.html

**Review by RyanSpies-NOAA — COMMENTED** [review]
Reviewed all code changes. Suggest one modification in `src/src_adjust_spatial_obs.py`

File: config/params_template.env (hunk 1 above)
Comment (robgpita-noaa) [reply config/params_template.env:43]: Good Catch! Thanks.

File: src/src_adjust_spatial_obs.py (hunk 2 above)
Comment (robgpita-noaa) [reply src/src_adjust_spatial_obs.py]: @RyanSpies-NOAA - I must've mis-understood the docs when I made this change. Reviewing the docs again, you are correct. I reverted back to `.to_crs`, ran a `fim_pipeline` test with one HUC, and everything checks out. 

The projection of the `water_edge_df` is set correctly to the `DEFAULT_FIM_PROJECTION` as verified by: 
1.) Running this script with `-debug` option (which writes `water_edge_df` to a `.parquet` file)
2.) Loading the `.parquet` file as a GDF
3.) Checking that the projection of the GDF matches the `DEFAULT_FIM_PROJECTION`

**Pull request merged**